### PR TITLE
Changes to add OCI feature to configmanagement gke hub feature

### PR DESCRIPTION
**Pull request by t-indumathy**
Updated DCL version to 1.28.0 to support OCI Config in google_gke_hub_feature_membership .
Added OCI config feature in google_gke_hub_feature_membership::configmanagement::config_sync .

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

```release-note:enhancement
anthos-fleet-management: added feature `configmanagement:config_sync:oci` to resource `google_gke_hub_feature_membership`.
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_gke_hub_feature_membership_test.go.erb [hunk 1]
```diff
@@ -573,6 +573,204 @@ resource "google_gke_hub_feature_membership" "feature_member" {
...
+      exemptable_namespaces = ["onetwothree", "fourfive"]
+      template_library_installed = true
+      referential_rules_enabled = true
+      log_denies_enabled = true
+    }
+  }
+  provider = google-beta
+}
+`, context)
+}
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/tests/resource_gke_hub_feature_membership_test.go.erb:772]: This is a challenging PR to review due to the tests being unable to run in CI. Were you able to run these tests locally and get them to pass. I've been trying to run them locally but running into failures on my end.

File: mmv1/third_party/terraform/tests/resource_gke_hub_feature_membership_test.go.erb [hunk 2]
```diff
@@ -573,6 +573,204 @@ resource "google_gke_hub_feature_membership" "feature_member" {
...
+    CheckDestroy: testAccCheckGKEHubFeatureDestroyProducer(t),
+    Steps: []resource.TestStep{
+      {
+        Config: testAccGkeHubFeatureMembership_gkehubFeatureAcmOciStart(context),
+        Check: resource.ComposeTestCheckFunc(
+          testAccCheckGkeHubFeatureMembershipPresent(t, fmt.Sprintf("tf-test-gkehub%s", context["random_suffix"]), "global", "configmanagement", fmt.Sprintf("tf-test1%s", context["random_suffix"])),
+        ),
+      },
+      {
+        ResourceName:      "google_gke_hub_feature_membership.feature_member",
```
Comment (ScottSuarez) [inline mmv1/third_party/terraform/tests/resource_gke_hub_feature_membership_test.go.erb:599]: This verify is looking for `google_gke_hub_feature_membership.feature_member` which doesn't exist in the configuration you've applied above.

You likely meant `google_gke_hub_feature_membership.feature_member_1`

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
Please ensure this test passes locally for you since the test is unable to run in CI.

File: mmv1/third_party/terraform/tests/resource_gke_hub_feature_membership_test.go.erb (hunk 2 above)
Comment (t-indumathy) [reply mmv1/third_party/terraform/tests/resource_gke_hub_feature_membership_test.go.erb:608]: Changed the TF resource ids to be inline with ResourceName.

File: mmv1/third_party/terraform/tests/resource_gke_hub_feature_membership_test.go.erb (hunk 1 above)
Comment (t-indumathy) [reply mmv1/third_party/terraform/tests/resource_gke_hub_feature_membership_test.go.erb]: Tested the changes in local and pasting the Acceptance Test results.

```
==> Checking that code complies with gofmt requirements...
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci
=== PAUSE TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci
=== CONT  TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci
--- PASS: TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci (956.07s)
PASS
ok  	github.com/hashicorp/terraform-provider-google-beta/google-beta	956.168s

```

**Comment by t-indumathy** [comment]
> Uploaded the Acceptance Test log.

[output.log](https://github.com/GoogleCloudPlatform/magic-modules/files/10253970/output.log)

**Comment by ScottSuarez** [comment]
I'm running into issues running the test locally. I don't think this is relevant to your changes specifically since the other tests are modeled in the same way. Could you try to rerun the test and confirm you aren't running into this issue. We may need to change the network setup to not use default since it seems there is collision there. I am suspecting it is auto-created on service activation but its strange that this issue has not be run into by other tests of this type.

```log
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci
=== PAUSE TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci
=== CONT  TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci
    provider_test.go:315: Step 1/6 error: Error running apply: exit status 1
        
        Error: Error creating Network: googleapi: Error 409: The resource 'projects/tf-test-gkehub1lfrghy8s0/global/networks/default' already exists, alreadyExists
        
          with google_compute_network.default,
          on terraform_plugin_test.tf line 57, in resource "google_compute_network" "default":
          57: resource "google_compute_network" "default" {
        
--- FAIL: TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci (130.07s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google-beta/google-beta 131.081s
FAIL
```

```log
---[ RESPONSE ]--------------------------------------
HTTP/2.0 409 Conflict
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Mon, 19 Dec 2022 23:25:36 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Google-Backends: [2002:a05:6681:c60b:b0:115:9977:8f7c]:9801,/bns/jy/borg/jy/bns/gce-control-esf-proxy/prod-global.esf-proxy/2,/bns/pf/borg/pf/bns/traffic-prod/shared-layer2-gfe/11,/bns/pf/borg/pf/bns/traffic-prod/shared-layer2-gfe/11,/bns/pnseab/borg/pnseab/bns/gfe-prod/blue-layer1-gfe.sea09s30/12
X-Google-Dos-Service-Trace: main:gce-control-esf-proxy-http-prod-global,main:shared-layer2-gfe
X-Google-Esf-Cloud-Client-Params: backend_service_name: "compute.googleapis.com" backend_fully_qualified_method: "compute.beta.NetworksService.Insert"
X-Google-Gfe-Handshake-Trace: GFE: /bns/pnseab/borg/pnseab/bns/gfe-prod/blue-layer1-gfe.sea09s30/12,Mentat oracle: [2002:ac4:c29b:0:b0:49d:3260:3a3d]:9801
X-Google-Gfe-Request-Trace: acseah11:443,pfod24:9801,/bns/jy/borg/jy/bns/gce-control-esf-proxy/prod-global.esf-proxy/2,pfod24:9801,acseah11:443
X-Google-Gfe-Response-Body-Transformations: chunked
X-Google-Gfe-Response-Code-Details-Trace: response_code_set_by_backend,response_code_set_by_backend
X-Google-Gfe-Service-Trace: gce-control-esf-proxy-http-prod-global,restricted-shared-layer2-grpc-aggregate
X-Google-Gfe-Version: 2.804.2
X-Google-Gslb-Service: gce-control-esf-proxy-http-prod-global
X-Google-Netmon-Label: /bns/jy/borg/jy/bns/gce-control-esf-proxy/prod-global.esf-proxy/2
X-Google-Security-Signals: FRAMEWORK=HTTPSERVER2,BUILD=GOOGLE3,BUILD_DEBUG=cl:485838367
X-Google-Service: gce-control-esf-proxy-http-prod-global,restricted-shared-layer2-grpc-aggregate
X-Google-Session-Info: GgQYECgLIAE6GBIWY29tcHV0ZS5nb29nbGVhcGlzLmNvbQ
X-Google-Shellfish-Status: CA0gBEBJ
X-Xss-Protection: 0

{
  "error": {
    "code": 409,
    "message": "The resource 'projects/tf-test-gkehubegs3jo0hq6/global/networks/default' already exists",
    "errors": [
      {
        "message": "The resource 'projects/tf-test-gkehubegs3jo0hq6/global/networks/default' already exists",
        "domain": "global",
        "reason": "alreadyExists",
        "debugInfo": "java.lang.Exception\n\tat com.google.cloud.control.common.publicerrors.PublicErrorProtoUtils.newErrorBuilder(PublicErrorProtoUtils.java:1960)\n\tat com.google.cloud.control.common.publicerrors.PublicErrorProtoUtils.createResourceAlreadyExistsError(PublicErrorProtoUtils.java:202)\n\tat com.google.cloud.control.frontend.action.MixerInsertActionExecutor$InnerHandler.ensureResourceDoesNotExist(MixerInsertActionExecutor.java:1016)\n\tat com.google.cloud.control.frontend.action.MixerInsertActionExecutor$InnerHandler.prepareMutationFlow(MixerInsertActionExecutor.java:848)\n\tat com.google.cloud.control.frontend.action.MixerInsertActionExecutor$InnerHandler.prepareMutationFlow(MixerInsertActionExecutor.java:701)\n\tat com.google.cloud.control.frontend.action.MixerMutationActionExecutor$InnerHandler.runAttempt(MixerMutationActionExecutor.java:1073)\n\tat com.google.cloud.control.frontend.action.MixerMutationActionExecutor$InnerHandler.runAttempt(MixerMutationActionExecutor.java:731)\n\tat com.google.cloud.cluster.metastore.RetryingMetastoreTransactionExecutor$1.runAttempt(RetryingMetastoreTransactionExecutor.java:79)\n\tat com.google.cloud.cluster.metastore.MetastoreRetryLoop.runHandler(MetastoreRetryLoop.java:499)\n\t...Stack trace is shortened.\n"
      }
    ]
  }
}
```

**Comment by t-indumathy** [comment]
I get the below error, when I don't create the `default` network explicitly:
`Error 400: Project \" tf-test-<<r-suffix>> \" has no network named \"default\"`
P.S: This error occurs for all the test cases of this resource.

Nonetheless, I've now created a separate dependency setup for this test case to avoid stepping on the other existing test cases. Below is the log for the acceptance test:
```
==> Checking that code complies with gofmt requirements...
go vet
go generate  ./...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci
=== PAUSE TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci
=== CONT  TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci
--- PASS: TestAccGkeHubFeatureMembership_gkehubFeatureAcmOci (994.78s)
PASS
ok  	github.com/hashicorp/terraform-provider-google-beta/google-beta	994.875s



```
Attaching the log file for acceptance test.
[output.log](https://github.com/GoogleCloudPlatform/magic-modules/files/10265499/output.log)

**Review by ScottSuarez — APPROVED** [review]

**Comment by ScottSuarez** [comment]
Thanks for the work !

ref: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstream/362156?buildTab=tests

**Pull request merged**